src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,228 @@
+import cx from 'classnames';
+
+export const TreeContext = React.createContext<
+  | {
+      selectedNodes?: Array<string>;
+      setSelectedNode?: (
+        node: Array<string> | ((prev: Array<string>) => Array<string>),
+      ) => void;
+      expandedNodes?: Array<string>;
+      setExpandedNodes?: (
+        node: Array<string> | ((prev: Array<string>) => Array<string>),
+      ) => void;
+      nodeDepth?: number;
+      parentNode?: FlatNode;
+    }
+  | undefined
+>(undefined);
+
+export type TreeData = {
+  subnodes: Array<TreeData>;
+  nodeId: string;
+  label?: string;
+  subLabel?: string;
+  isExpanded?: boolean;
+  isDisabled?: boolean;
+};
+
+type FlatNode = {
+  data: TreeData;
+  id: string;
+  depth: number;
+  subNodeIds: string[];
+  parent?: FlatNode;
+};
+
+export type TreeProps = {
+  /**
+   * Node renderer.
+   */
+  nodeRenderer?: (props: TreeData) => JSX.Element;
+  /**
+   * Items inside tree.
+   */
+  data?: Array<TreeData>;
+  /**
+   * Node ids of expanded nodes.
+   * If undefined, expanding and collapsing nodes will be handled automatically.
+   */
+  expandedNodes?: Array<string>;
+  /**
+   * Node ids of selected nodes.
+   * If undefined, selecting nodes will be handled automatically.
+   */
+  selectedNodes?: Array<string>;
+} & CommonProps;
+
+/**
+ * @example
+  <Tree
+    data={data}
+    nodeRenderer={(props) => (
+      <TreeNode
+        nodeId={props.nodeId}
+        label={props.label}
+        sublabel={props.subLabel}
+        subNodes={props.subnodes}
+        onNodeExpanded={onNodeExpanded}
+        onNodeSelected={onSelectedNodeChange}
+        isDisabled={props.isDisabled}
+        nodeCheckbox={<Checkbox variant='eyeball' checked={true} />}
+        onNodeCheckboxSelected={onCheckboxSelected}
+        icon={<SvgPlaceholder />}
+      />
+    )}
+    expandedNodes={expandedNodes}
+    selectedNodes={selectedNodes}
+    {...args}
+  />
+ */
+export const Tree = (props: TreeProps) => {
+  const {
+    data,
+    className,
+    nodeRenderer,
+    expandedNodes,
+    selectedNodes,
+    ...rest
+  } = props;
+  useTheme();
+
+  const treeRef = React.useRef<HTMLUListElement>(null);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLUListElement>) => {
+    const nodes: Array<HTMLElement> = Array.from(
+      treeRef.current?.querySelectorAll('.iui-tree-node:not(.iui-disabled)') ||
+        [],
+    );
+
+    if (!nodes.length) {
+      return;
+    }
+
+    let newIndex = -1;
+    const currentIndex = nodes.findIndex(
+      (node) => node === treeRef.current?.ownerDocument.activeElement,
+    );
+
+    const expander = nodes[currentIndex].querySelector(
+      '.iui-tree-node-content-expander-icon',
+    ) as HTMLElement;
+
+    switch (event.key) {
+      case 'ArrowDown': {
+        newIndex = currentIndex + 1;
+        break;
+      }
+      case 'ArrowUp': {
+        newIndex = currentIndex - 1;
+        break;
+      }
+      case 'ArrowLeft':
```

Comment:
Hm, Maybe we can handle left and right like in `MenuItem`: call `onExpand` if it has subrows, if not we can focus its parent. Maybe we can use parent `nodeId` to focus it.

Reply:
I changed left to either collapse node or go to parent node if already collapsed